src/bun.js/bindings/BunObject.cpp

```diff
@@ -188,7 +197,19 @@ JSC_DEFINE_HOST_FUNCTION(functionConcatTypedArrays, (JSGlobalObject * globalObje
 
     auto arrayValue = callFrame->uncheckedArgument(0);
 
-    return flattenArrayOfBuffersIntoArrayBuffer(globalObject, arrayValue);
+    size_t maxLength = SIZE_MAX;
+    auto arg1 = callFrame->argument(1);
+    if (!arg1.isUndefinedOrNull() && arg1.toNumber(globalObject) != INFINITY) {
```

Comment:
`toNumber` calls `[Symbol.toPrimitive]`, which means it can throw.

Here's what we do in Buffer:

https://github.com/oven-sh/bun/blob/9a352b7d1202c0ef2e7746f887e9b6c7e15d34c3/src/bun.js/bindings/JSBuffer.cpp#L1290

I suggest something like this:

```suggestion
    if (!arg1.isUndefined() && arg1.isNumber()) {
       double number = arg1.toNumber(globalObject);
       RETURN_IF_EXCEPTION(scope, {});
       if (std::isnan(number) || number < 0) {
          throwRangeError(lexicalGlobalObject, scope, "Maximum length must be >= 0"_s);
          return {};
       }
       maxLength = std::min(std::numeric_limits<uint32_t>::max(), static_cast<size_t>(maxLength));
```